packages/transaction-controller/src/TransactionController.ts

```diff
@@ -1618,6 +1667,9 @@ export class TransactionController extends BaseControllerV1<
    * @param opts.key - The hex address (or constant) pertaining to the chainId
    * @returns Mutex instance for the given chainId and key pair
    */
+  // This should live in the Multichain helper, except that it's needed in this
+  // controller context as well. It's possible to move this, but just highlighting the coupling.
+  // MultichainHelper is really TrackingHelper and would contain the global trackers.
```

Comment:
That was absolutely my intent, to encapsulate the tracking and any manipulation of the map and the creation of the helpers.

I think the name is introducing confusion here so feel free to call it `MultichainTrackingHelper` if that is more explicit.

Reply:
https://github.com/MetaMask/core/pull/3896/commits/c1f4dc586cdd54ba013fef9c9d1e802286b6f0b6